Ta load capability

dsemenets opened the pull request.

dsemenets committed:
libteec: Initial version of libteec implementation

Code was taken from the optee repo:
https://github.com/OP-TEE/optee_client.git
Tag: 3.21.0
Original code style was kept.

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

dsemenets committed:
libckteec: Initial commit for libckteec

Code was taken from the optee repo:
https://github.com/OP-TEE/optee_client.git
Tag: 3.21.0
Original code style was kept.

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev commented on tee_supplicant/CMakeLists.txt:
```diff
@@ -8,5 +8,6 @@ target_include_directories(TEE_SUPPLICANT INTERFACE include)
 zephyr_include_directories(include)
-zephyr_include_directories(../public)
\ No newline at end of file
+zephyr_include_directories(../public)
```
please remove extra line

dsemenets replied:
This line is not my change

oleksiimoisieiev commented on tee_supplicant/include/teec_ta_load.h:
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2014, STMicroelectronics International N.V.
```
Are you sure it's correct Copyright?

dsemenets replied:
Corrected

oleksiimoisieiev commented on tee_supplicant/include/teec_ta_load.h:
```diff
@@ -0,0 +1,58 @@
+	size_t ta_size;
+};
+
+void TEEC_SetTATable(struct ta_table *table);
```
This requires comment as it is in LoadSecureModule

dsemenets replied:
Done

oleksiimoisieiev replied:
+

oleksiimoisieiev commented on tee_supplicant/src/tee_supplicant.c:
```diff
@@ -21,6 +22,25 @@ LOG_MODULE_REGISTER(tee_supplicant);
+	void *p;
+	size_t size;
+	bool registered;
+	int fd;
```
why do you need fd here?

dsemenets replied:
removed

oleksiimoisieiev commented on tee_supplicant/src/tee_supplicant.c:
```diff
@@ -62,10 +88,134 @@ static int send_response(const struct device *dev, struct tee_supp_response *rsp
+	sz = MEMREF_SIZE(params + idx);
+	offs = MEMREF_SHM_OFFS(params + idx);
+	if ((sz + offs) < sz)
+		return -1;
```
Please return at least errno on error

dsemenets replied:
Ok

oleksiimoisieiev commented on tee_supplicant/src/tee_supplicant.c:
```diff
@@ -124,4 +274,26 @@ int tee_supp_init(const struct device *dev)
+
+	if (e) {
+		LOG_ERR("pthread_mutex_lock: %d", e);
+//		LOG_ERR("terminating...");
```
please no comments

dsemenets replied:
Done

oleksiimoisieiev commented on tee_supplicant/src/tee_supplicant.c:
```diff
@@ -124,4 +274,26 @@ int tee_supp_init(const struct device *dev)
+
+	if (e) {
+		LOG_ERR("pthread_mutex_unlock: %d", e);
+//		LOG_ERR("terminating...");
```
please no comments

dsemenets replied:
Done

oleksiimoisieiev commented on tee_supplicant/src/teec_ta_load.c:
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2014, STMicroelectronics International N.V.
```
copyright?

dsemenets replied:
Fixed

oleksiimoisieiev commented on tee_supplicant/CMakeLists.txt:
```diff
@@ -8,5 +8,6 @@ target_include_directories(TEE_SUPPLICANT INTERFACE include)
 
 zephyr_library()
 zephyr_library_sources(src/tee_supplicant.c)
+zephyr_library_sources(src/teec_ta_load.c)
```
Commit descrition

oleksiimoisieiev commented on tee_supplicant/src/teec_ta_load.c:
```diff
@@ -0,0 +1,77 @@
+
+void TEEC_SetTATable(struct ta_table *table)
+{
+	ta_table = table;
```
if (table != NULL)?

dsemenets replied:
Why?

oleksiimoisieiev replied:
+

oleksiimoisieiev commented on tee_supplicant/src/teec_ta_load.c:
```diff
@@ -0,0 +1,77 @@
+		     destination->clockSeqAndNode[7]);
+
+	for (t = ta_table; t->uuid != NULL; t++)
+		if (!strcmp(uuid, t->uuid)) {
```
please prefer using strncmp

dsemenets replied:
Ok

oleksiimoisieiev commented on tee_supplicant/src/teec_ta_load.c:
```diff
@@ -0,0 +1,77 @@
+	struct ta_table *t;
+	char uuid[PATH_MAX] = { 0 };
+
+	if (!ta || !ta_table)
```
|| !ta_size

dsemenets replied:
Fixed

oleksiimoisieiev commented on tee_supplicant/src/tee_supplicant.c:
```diff
@@ -41,7 +41,7 @@ union tee_supp_msg {
 static int receive_request(const struct device *dev, struct tee_supp_request *ts_req)
 {
 	LOG_ERR("Receiving request...");
```
Please try to avoid this commit

dsemenets replied:
Actually it is not my change but Ok

oleksiimoisieiev commented:
Add commit message and +1 
`Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>`

dsemenets committed:
tee_supplicant: Load TA

Initial version of tee_supplicant.
This version processes only TA load request.
TA load function finds appropriate TA in TA table and moves it
to shared memory.

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Co-authored-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Tested-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>

dsemenets committed:
libckteec: Fixed dependency for libckteec

Fixed missprint in Kconfig

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babcbuk <volodymyr_babchuk@epam.com>

dsemenets committed:
Added file from official optee-client

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
